sdk/identity/azure-identity/src/samples/Azure-Cache-For-Redis/Jedis/Azure-AAD-Authentication-With-Jedis.md

```diff
@@ -78,6 +78,27 @@ System.out.println(jedis.get("Az:key"));
 
 // Close the Jedis Client
 jedis.close();
+
+
+private static String extractUsernameFromToken(String token) {
+    String[] parts = token.split("\\.");
+    String base64 = parts[1];
+
+    switch (base64.length() % 4) {
```

Comment:
is there no standard java lib for parsing tokens like Microsoft.IdentityModel.JsonWebTokens in .NET?

Reply:
There are 3rd party libraries. 
but they bring in the issues of being exposed to CVEs, stale versions (if not updated) and dependency conflicts.
For this operation, native Java SDK is reliable.